### DISCUSSION TITLE
Sonoff ifan04-H maybe not operating as expected?

### DISCUSSION THREAD

**Discussion by sotiris-bos** [arendst/Tasmota#16402] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I have a Sonoff ifan04-H model. I have replaced the two capacitors with two smaller ones, a 1uF and a 1.5uF capacitor because the included caps had too high of a capacitance and span my fan too fast. However, the difference between fanspeed 1 and fanspeed 2 is very large, the fan consumes 7w when running at fanspeed 1 with cap 1uF, and 32w+ when running at fanspeed 2 with cap 1.5uF.

I am not sure if this is related to the caps or tasmota but I want to raise this issue: The original Sonoff firmware activates both caps when fanspeed 2 is enabled. The tasmota firmware is supposed to only activate one of the caps when it is in either fanspeed 1 or fanspeed 2.

Could this be a case where the firmware does not work as expected and activates both caps at the same time on ifan04-H? I also own a Sonoff ifan03, installed on a second identical fan, with standard caps (2.5uF for fanspeed 1 and 3uF for fanspeed 2) and the rotation speed of ifan04 with fanspeed 2 (1.5uF supposedly) and ifan03 at fanspeed 1 (2.5uF) is quite similar, but I don't think it is identical.

Is there anything I can do to make sure that only one of the relays and hence only one of the caps are providing power to the fan? Any logs or files I can share with you to diagnose?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff ifan04-H
- [x] Tasmota binary firmware version number used: 12.0.2
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Tasmotizer
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
14:30:36.348 CMD: Backlog Template; Module; GPIO 255
14:30:36.403 MQT: stat/tasmota_77E276/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
14:30:36.615 MQT: stat/tasmota_77E276/RESULT = {"Module":{"71":"Sonoff iFan03"}}
14:30:36.869 MQT: stat/tasmota_77E276/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"3200":"Serial Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3232":"Serial Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"256":"Relay_i1"},"GPIO10":{"512":"Buzzer_i"},"GPIO12":{"226":"Relay3"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"225":"Relay2"},"GPIO15":{"227":"Relay4"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
14:32:08.780 CMD: Backlog Rule1; Rule2; Rule3
14:32:08.848 MQT: stat/tasmota_77E276/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:09.062 MQT: stat/tasmota_77E276/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:09.312 MQT: stat/tasmota_77E276/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:26:41.001 CMD: Status 0
14:26:41.008 MQT: stat/tasmota_77E276/STATUS = {"Status":{"Module":71,"DeviceName":"Sotiris Fan","FriendlyName":["Sotiris Fan"],"Topic":"tasmota_77E276","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
14:26:41.015 MQT: stat/tasmota_77E276/STATUS1 = {"StatusPRM":{"Baudrate":9600,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Power On","Uptime":"0T00:00:32","StartupUTC":"2022-08-16T13:26:09","Sleep":50,"CfgHolder":4617,"BootCount":29,"BCResetTime":"2022-08-09T18:46:30","SaveCount":217,"SaveAddress":"FB000"}}
14:26:41.024 MQT: stat/tasmota_77E276/STATUS2 = {"StatusFWR":{"Version":"12.0.2(tasmota)","BuildDateTime":"2022-06-20T12:41:03","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"337/699"}}
14:26:41.029 MQT: stat/tasmota_77E276/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["iot",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006000","00004000","00000000"]}}
14:26:41.042 MQT: stat/tasmota_77E276/STATUS4 = {"StatusMEM":{"ProgramSize":619,"Free":384,"Heap":23,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14325E","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","14000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
14:26:41.050 MQT: stat/tasmota_77E276/STATUS5 = {"StatusNET":{"Hostname":"tasmota-77E276-0630","IPAddress":"10.0.3.43","Gateway":"10.0.3.1","Subnetmask":"255.255.255.0","DNSServer1":"10.0.3.1","DNSServer2":"0.0.0.0","Mac":"70:03:9F:77:E2:76","Webserver":2,"HTTP_API":1,"WifiConfig":2,"WifiPower":17.0}}
14:26:41.056 MQT: stat/tasmota_77E276/STATUS6 = {"StatusMQT":{"MqttHost":"10.0.3.10","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_77E276","MqttUser":"hass","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
14:26:41.064 MQT: stat/tasmota_77E276/STATUS7 = {"StatusTIM":{"UTC":"2022-08-16T13:26:41","Local":"2022-08-16T14:26:41","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"05:44","Sunset":"20:04"}}
14:26:41.071 MQT: stat/tasmota_77E276/STATUS10 = {"StatusSNS":{"Time":"2022-08-16T14:26:41"}}
14:26:41.079 MQT: stat/tasmota_77E276/STATUS11 = {"StatusSTS":{"Time":"2022-08-16T14:26:41","Uptime":"0T00:00:32","UptimeSec":32,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","FanSpeed":0,"Wifi":{"AP":1,"SSId":"iot","BSSId":"8A:8A:20:D7:7C:E9","Channel":6,"Mode":"11n","RSSI":44,"Signal":-78,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:30:36.869 MQT: stat/tasmota_77E276/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"3200":"Serial Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3232":"Serial Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"256":"Relay_i1"},"GPIO10":{"512":"Buzzer_i"},"GPIO12":{"226":"Relay3"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"225":"Relay2"},"GPIO15":{"227":"Relay4"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
14:31:18.408 MQT: tele/tasmota_77E276/STATE = {"Time":"2022-08-16T14:31:18","Uptime":"0T00:05:09","UptimeSec":309,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","FanSpeed":0,"Wifi":{"AP":1,"SSId":"iot","BSSId":"8A:8A:20:D7:7C:E9","Channel":6,"Mode":"11n","RSSI":42,"Signal":-79,"LinkCount":1,"Downtime":"0T00:00:03"}}
14:31:50.769 CMD: Backlog Rule1; Rule2; Rule3
14:31:50.843 MQT: stat/tasmota_77E276/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:31:51.055 MQT: stat/tasmota_77E276/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:31:51.305 MQT: stat/tasmota_77E276/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:08.780 CMD: Backlog Rule1; Rule2; Rule3
14:32:08.848 MQT: stat/tasmota_77E276/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:09.062 MQT: stat/tasmota_77E276/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:09.312 MQT: stat/tasmota_77E276/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:32:59.445 MQT: stat/tasmota_77E276/RESULT = {"SerialLog":{"0":{"Active":"0"}}}
14:32:59.456 MQT: stat/tasmota_77E276/RESULT = {"WebLog":4}
14:32:59.461 SRC: Backlog
14:32:59.464 CMD: Grp 0, Cmd 'MQTTLOG', Idx 1, Len 1, Pld 0, Data '0'
14:32:59.470 MQT: stat/tasmota_77E276/RESULT = {"MqttLog":0}
14:32:59.474 SRC: Backlog
14:32:59.477 CMD: Grp 0, Cmd 'SYSLOG', Idx 1, Len 1, Pld 0, Data '0'
14:32:59.482 MQT: stat/tasmota_77E276/RESULT = {"SysLog":{"0":{"Active":"0"}}}
14:32:59.487 SRC: Backlog
14:32:59.489 CMD: Grp 0, Cmd 'LOGHOST', Idx 1, Len 1, Pld 1, Data '1'
14:32:59.491 CFG: CR 337/699, Busy 0
14:32:59.495 MQT: stat/tasmota_77E276/RESULT = {"LogHost":""}
14:32:59.502 SRC: Backlog
14:32:59.504 CMD: Grp 0, Cmd 'LOGPORT', Idx 1, Len 3, Pld 514, Data '514'
14:32:59.508 MQT: stat/tasmota_77E276/RESULT = {"LogPort":514}
14:32:59.514 SRC: Backlog
14:32:59.517 CMD: Grp 0, Cmd 'TELEPERIOD', Idx 1, Len 3, Pld 300, Data '300'
14:32:59.523 MQT: stat/tasmota_77E276/RESULT = {"TelePeriod":300}
14:33:00.425 MQT: tele/tasmota_77E276/STATE = {"Time":"2022-08-16T14:33:00","Uptime":"0T00:06:51","UptimeSec":411,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","FanSpeed":0,"Wifi":{"AP":1,"SSId":"iot","BSSId":"8A:8A:20:D7:7C:E9","Channel":6,"Mode":"11n","RSSI":40,"Signal":-80,"LinkCount":1,"Downtime":"0T00:00:03"}}
14:33:00.651 CFG: Saved to flash at FA, Count 218, Bytes 4096
14:33:01.365 HTP: Main Menu
14:33:02.891 SRC: WebGui from 172.16.0.2
14:33:02.893 SRC: WebGui
14:33:02.895 CMD: Grp 0, Cmd 'FANSPEED', Idx 1, Len 1, Pld 1, Data '1'
14:33:02.903 MQT: stat/tasmota_77E276/RESULT = {"FanSpeed":1}
14:33:03.654 CFG: Saved to flash at F9, Count 219, Bytes 4096
14:33:08.649 CFG: Saved to flash at F8, Count 220, Bytes 4096
14:33:11.385 WIF: Sending Gratuitous ARP
14:33:12.786 WIF: Checking connection...
14:33:32.777 WIF: Checking connection...
14:33:52.766 WIF: Checking connection...
14:34:03.299 SRC: WebGui from 172.16.0.2
14:34:03.301 SRC: WebGui
14:34:03.303 CMD: Grp 0, Cmd 'FANSPEED', Idx 1, Len 1, Pld 2, Data '2'
14:34:03.311 MQT: stat/tasmota_77E276/RESULT = {"FanSpeed":2}
14:34:03.662 CFG: Saved to flash at F7, Count 221, Bytes 4096
14:34:11.383 WIF: Sending Gratuitous ARP
14:34:11.588 HTP: Console
14:34:12.791 WIF: Checking connection...
14:34:32.786 WIF: Checking connection...
14:34:52.764 WIF: Checking connection...

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
I would expect fanspeed 1 and fanspeed 2 to be more similar to each other. My old controller was running a 1.5uF cap for its fanspeed 2 and the fan was rotating much more slowly than the Sonoff ifan04 with tasmota and 1.5uF capacitor for its fanspeed 2.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
Tasmota controls the same relays as the original firmware.

See source code regarding fanspeed control:
```
uint8_t GetFanspeed(void)
{
  if (ifan_fanspeed_timer) {
    return ifan_fanspeed_goal;                     // Do not show sequence fanspeed
  } else {
    /* Fanspeed is controlled by relay 2, 3 and 4 as in Sonoff 4CH.
      000x = 0
      001x = 1
      011x = 2
      101x = 3 (ifan02) or 100x = 3 (ifan03)
    */
    uint8_t fanspeed = (uint8_t)(TasmotaGlobal.power &0xF) >> 1;
    if (fanspeed) { fanspeed = (fanspeed >> 1) +1; }  // 0, 1, 2, 3
    return fanspeed;
  }
}
```

**Comment by sotiris-bos** [comment]
> Tasmota controls the same relays as the original firmware.
> 
> See source code regarding fanspeed control:
> 
> ```
> uint8_t GetFanspeed(void)
> {
>   if (ifan_fanspeed_timer) {
>     return ifan_fanspeed_goal;                     // Do not show sequence fanspeed
>   } else {
>     /* Fanspeed is controlled by relay 2, 3 and 4 as in Sonoff 4CH.
>       000x = 0
>       001x = 1
>       011x = 2
>       101x = 3 (ifan02) or 100x = 3 (ifan03)
>     */
>     uint8_t fanspeed = (uint8_t)(TasmotaGlobal.power &0xF) >> 1;
>     if (fanspeed) { fanspeed = (fanspeed >> 1) +1; }  // 0, 1, 2, 3
>     return fanspeed;
>   }
> }
> ```

Huh. I found this post that let me believe that the author made a Pull Request that got accepted: https://www.breznet.com/sonoff-ifan03-home-assistant/

The post claims that the original handling (original firmware) is wrong and that the relays should not be powered on concurrently.

**Comment by barbudor** [comment]
@sotiris-bos He submitted and merged the PR on it's own fork. Can't any clue that he ever been submitted to arendst/asmota
Nevertheless I have no idea if his claim is correct or not

**Comment by arendst** [comment]
ifan02 and ifan03 have been around for years. I have them both and they control the fan exactly as advertised using tasmota so I see no reason why to change stable code other than the fact that ifan04 may work differently then the previous ones.

Let other ifan04 users speak if they find fan control wlrks different on ifan04 compared to the previous ones.

**Reply by dphuk** [reply]
Hi, I flashed an iFan04-H today (UK 240V) and after loading the iFan04-H template I noticed that low worked as expected, medium was very fast and high was exactly the same as medium. After looking at the code and the links above, I modified and compiled the Tasmota code to replace "xdrv_22_sonoff_ifan.ino" on line 30 with:
const uint8_t kIFan03Speed[MAX_FAN_SPEED +2] = { 0x00, 0x01, 0x02, 0x04, 0x05, 0x06 };
This now sets 3 different speeds and works correctly. I've not changed any of the hardware, nor do I have any previous models to compare this with I'm afraid.

**Reply by HonzaCZ** [reply]
Hello, I have iFan04-H (230 V) and CasaFan 9313210 MIRAGE, and I can confirm, that [this patch](https://github.com/brezlord/Tasmota/pull/1/commits/619228089d26fdc255e8199de0930d419d52304e) works well and makes sense - at least for this case.

With standard Tasmota (14.1) there is no difference between speed level 2 and 3, and there is full power startup procedure only for speed 1.

With mentioned patch there is useful speed 2. All 3 speeds are now comparable to factory controller (CasaFan 85220 FB-FNK Powerboat). And in addition there is full power startup sequence also for speed 2.

Unfortunately I have no iFan02 or -03, so I can not compare settings and behavior for this devices. But for iFan04-H I would recommend adding some option to involve modified code. E.g. exclusive driver for iFan04, some setOption or at least adding constant for easy build-time configuration.

Something like:
```c
#ifdef USE_SONOFF_IFAN4_ALT
    const uint8_t kIFan03Speed[MAX_FAN_SPEED +2] = { 0x00, 0x01, 0x02, 0x04, 0x05, 0x06 };
    const uint8_t kIFan03Sequence[MAX_FAN_SPEED][MAX_FAN_SPEED] = {{0, 4, 5, 3}, {0, 1, 2, 3}, {0, 1, 2, 3}, {0, 4, 5, 3}};
#else
    const uint8_t kIFan03Speed[MAX_FAN_SPEED +2] = { 0x00, 0x01, 0x03, 0x04, 0x05, 0x06 };
    const uint8_t kIFan03Sequence[MAX_FAN_SPEED][MAX_FAN_SPEED] = {{0, 2, 2, 2}, {0, 1, 2, 4}, {1, 1, 2, 5}, {4, 4, 5, 3}};
#endif
```

**Reply by ferbulous** [reply]
Glad I found this fix for speed 2 going too fast with ifan04
This has been bugging me ever since it's installed and I couldn't figure out what's the best way to deal with this if I needed to change the capacitor that came with the ifans.
Will try it as well on ifan03

**Reply by arendst** [reply]
Let me find a way to easily add this fix ...

**Comment by arendst** [comment, marked as answer]
Latest dev branch v14.1.0.4 contains a fix for iFan04-H.

To use it load and activate below template:
```
{"NAME":"Sonoff iFan04-H","GPIO":[32,3200,5735,3232,0,0,256,512,226,320,225,227,0,0],"FLAG":0,"BASE":71}
```

**Reply by sotiris-bos** [reply]
Flashed my ifan04-H, now fanspeed 1 is 7w and fanspeed 2 is 13w, so works as expected!

Thank you!

**Reply by HonzaCZ** [reply]
> To use it load and activate below template:

Works pretty well! Thank you!

**Comment by xoxys** [comment]
I have a slightly different problem. For my fan, even step 1 is way too fast compared to the original controller. Is there any way to customize the fan speed for the presets?

**Reply by xoxys** [reply]
@sotiris-bos which capacitors did you buy? Im struggling to find caps that still fit into the case.

**Comment by joluinfante** [comment]
Hello. I converted two of my four iFAN04-H fans to Tasmota. And, among many other problems (which seem unsolvable, such as the Wi-Fi not responding in time to turn the light on or off), I encountered this one, which I find terrible: with the remote, I turn on one fan, and the one in the next room turns on as well. And that didn't happen with the original software. According to chatgpt, Tasmota does what it can with fairly limited hardware like the iFAN04 (I thought it was good quality hardware; it's present in many fans). Before deciding to build a Zigbee remote for the fan, I'd like to ask the community if my purchase of the four (five, because I bought one more for the Tasmota migration) was truly a "bad" decision, or if I simply let myself be guided by the AI ​​(mind you, I've been working on this for several months, making small adjustments to timeouts, etc., but the RF issue is something that worries me a lot).

**Reply by arendst** [reply]
As far as I remember every sonoff remote control has a unique address and needs to be paired with the RF hardware of the ifan. There is no tasmota involved with initial RF decodong. 

Tasmota does however decode received data send over serial to control the fan.

I suspect you've learned the same remote control to both ifans and hence they operate in sync with your remote control. Try to "unlearn" one ifan and let it "learn" the other remote control to get them out of sync.

**Reply by joluinfante** [reply]
Hi. I liked your perspective. It's true, I turned off and "reassociated" each remote, but I didn't unassociate them. And that would explain why one remote works differently than the other. I'll test it today when I get home and let you know. This possibility makes me feel a bit more at ease. Because really, apart from the ifan04 (and some Chinese rebranded versions), there isn't another product that allows you to control fans. And the alternative is to build a prototype with "pieces" that include logic in Home Assistant (or some other product) for the fan's operation. And then there are the aesthetic issues. I've really gotten used to the proprietary controls of the ifan04, and besides, I've already made the entire investment.

**Comment by joluinfante** [comment]
Hello. I unlearned the controller, relearned it with the second fan powered off, and it's still turning on both fans. This is happening with both IFA-04-H controllers I converted to Tasmota (the others, with the original software, aren't affected, at least not by this problem).

**Reply by sfromis** [reply]
Did you relearn on both fans, with their separate remotes? AFAIK, each remote has a fixed ID, which is remembered by the fan part after pairing. Hence you need to separately pair each remote with a specific fan. And Tasmota is indeed not involved in the pairing process.

Beware that each fan can hold info for being paired with up to 10 remotes, meaning that you have to explicitly unpair a remote doing too much from the device it should not be controlling. And you should not power both fans on at the same time when pairing, as they will go into pairing mode when they are powered on, each time.

**Comment by joluinfante** [comment]
That's right. You were right. I have (because experience taught me) a small circuit breaker to cut the power, mounted on the ceiling, right next to each fan. So, each reconnection required turning off both fans, turning on the one I was going to pair with the remote, and then doing the same for each one. Finally, I got it working. You need a bit of patience (and agility, because of the stool). Now each RF fan works independently. The only remaining issue is that sometimes the iFAN04 goes into sleep mode and doesn't respond to a Zigbee key (which communicates with Home Assistant, and there an automation runs to tell the fan to "alternate the light state"). It's like something goes into sleep mode; I have to press it once, wait a few seconds, press it again, and maybe press it once more. Then it works fine for a while, then goes into sleep mode again. I followed tons of AI recipes, adjusting this or that parameter, and it's just not working smoothly. The problem is that the little button on the wall remote is too small. That's why I added the larger button. Is there a "definitive" solution that won't fail? I have three TP-Link mesh systems distributed throughout the house, with each fixed device connected to one of them. I've already checked the mesh logs and found nothing. I understand that you have to tell the ifan04 "never go to sleep," but I haven't been able to get it to work. And I see there are many parameters. On both systems, I get this message frequently appearing in the Tasmota console, which would indicate that it should never go to sleep: 13:41:16.213 MQT: tele/ifan_living/STATE = {"Time":"2026-02-11T13:41:16","Uptime":"0T00:08:44","UptimeSec":524,"Heap":25,"SleepMode":"Dynamic","Sleep":0,"LoadAvg":378,"MqttCount":1,"POWER1":"OFF","FanS speed":0,"Wifi":{"AP":1,"SSId":"home-IoT","BSSId":"16:EF:15:4B:C6:46","Channel":10,"Mode":"11n","RSSI":100,"Signal":-47,"LinkCount":1,"Downtime":"0T00:00:08"}}

**Reply by sfromis** [reply]
Not clear what exactly you think of with "sleep". There is no Tasmota setup for device sleeping, outside of the `DeepSleep` feature. You know it if you are using that, and not at all suitable for a device you want to control on the push of a button. Also depends on how you are communicating with it, like using MQTT.

The STATE message you quoted indicates that it was booted 524 seconds ago, hence no indication of it running stably. Of course, if you know why it as booted a short time ago..... If you were power cycling to get it alive again, there can be many reasons for being unavailable, just not "sleeping". One risk factor is a Wifi router not working well with low-activity devices running Tasmota.

I'm assuming that "Zigbee key" is just a button and you are sending commands from Home Assistant for control. If using the preferable MQTT, HA should also have a state for device being online or not.
